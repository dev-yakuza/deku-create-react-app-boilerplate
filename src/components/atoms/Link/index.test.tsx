import { render, screen, fireEvent } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import {
  BrowserRouter,
  unstable_HistoryRouter as HistoryRouter,
} from 'react-router-dom'

import { Link } from '.'

describe('<Link />', () => {
  it('Rendered well', async () => {
    const { container } = render(
      <BrowserRouter>
        <Link to="/posts/1">test link</Link>
      </BrowserRouter>,
    )

    expect(screen.queryByText('test link')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('Go to the link URL when the link is clicked', () => {
    const mockPush = jest.fn()
    const history = createMemoryHistory({ initialEntries: ['/'] })
    history.push = mockPush

    render(
      <HistoryRouter history={history}>
        <Link to="/posts/1">test link</Link>
      </HistoryRouter>,
    )

    fireEvent.click(screen.getByText('test link'))
    expect(mockPush.mock.calls[0][0].pathname).toBe('/posts/1')
  })
})
