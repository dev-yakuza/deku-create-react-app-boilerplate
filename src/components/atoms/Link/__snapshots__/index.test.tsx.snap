// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Rendered well 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: initial;
}

<div>
  <a
    class="emotion-0"
    href="/posts/1"
  >
    test link
  </a>
</div>
`;
